packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3287,9 +3287,31 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
   @mustCallSuper
   void didChangeDependencies() {
     assert(_active); // otherwise markNeedsBuild is a no-op
+    _debugCheckOwnerBuildTargetExists('didChangeDependencies');
     markNeedsBuild();
   }
 
+  void _debugCheckOwnerBuildTargetExists(String methodName) {
+    assert(() {
+      if (owner._debugCurrentBuildTarget == null) {
+        throw new FlutterError(
+          '$methodName for ${widget.runtimeType} was called at an '
+          'inappropriate time.\n'
+          '\n'
```

Comment:
generally we don't include blank lines in the error messages; we rely on newlines alone being paragraph breaks.